flight/drone/drone.py

```diff
@@ -0,0 +1,186 @@
+            self._optical_flow.flow_comp_m_x = message.quality
+            self._optical_flow.flow_comp_m_y = message.flow_comp_m_y
+            self._optical_flow.quality = message.quality
+            self._optical_flow.ground_distance = message.ground_distance
+
+            # Notify all observers of new message (with new value)
+            #   Note that argument `cache=False` by default so listeners
+            #   are updaed with every new message
+            self.notify_attribute_listeners('optical_flow', self._optical_flow)
+
+    @property
+    def optical_flow(self):
+        """Get data from the optical flow sensor.
+
+        Notes
+        -----
+        See optical_flow_attribute.py for what kind of data you can get.
+        """
+        return self._optical_flow
+
+    @property
+    def id(self):
+        """Get the drone's id"""
+        return self._id
+
+    @id.setter
+    def id(self, identifier):
+        """Set the drone's id."""
+        self._id = identifier
+
+    def set_attitude(self, roll, pitch, yaw, thrust):
+        """Set the drones attitude.
+
+        Parameters
+        ----------
+        roll : double
+            The roll angle.
+        pitch : double
+            The pitch angle.
+        yaw : double
+            The yaw rate.
+        thrust : double between 0 and 1
+            The thrust value.
+
+        Notes
+        -----
+        If thrust < 0.5, the drone will lose altitude
+        If thrust == 0.5, the drone will retain its altitude
+        If thrust > 0.5, the drone will gain altitude
+        """
+        msg = self._make_attitude_message(roll, pitch, yaw, thrust)
+        self.send_mavlink(msg)
+
+    def send_velocity(self, north, east, down):
+        """Send velocity to the drone.
+
+        Parameters
+        ----------
+        north : double
+        east : double
+        down : double
+
+        Notes
+        -----
+        This method used the NED coordinate system. Of note is that sending a
+        positive value for down will make the drone lose altitude.
+        """
+        msg = self._make_velocity_message(north, east, down)
+        self.send_mavlink(msg)
+
+    def _make_velocity_message(self, north, east, down):
+        """Construct a mavlink message for sending velocity.
+
+        Parameters
+        ----------
+        north : double
+        east : double
+        down : double
+
+        Returns
+        -------
+        MAVLink_message (a DroneKit object)
+            Message which moves the drone at a certain velocity.
+
+        see http://python.dronekit.io/examples/guided-set-speed-yaw-demo.html#send-global-velocity
+        """
+        return self.message_factory.set_position_target_global_int_encode(
+            0,       # time_boot_ms (not used)
+            0, 0,    # target system, target component
+            mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT_INT, # frame
+            0b0000111111000111, # type_mask (only speeds enabled)
```

Comment:
What's the reasoning for removing the Bitmask enum?

Reply:
They got lost in the shuffle. I have added a MavBitmasks class in the constant folder and made a docstring that explains the meaning of the bitmasks and provides links to the fields the bitmask controls.